### PR TITLE
fix(strvals): port #3912, #4142, #4682, and #5151 to Helm 3

### LINKED ISSUE #6940: helm 3.0.0-rc.3 --set-string does not work with booleans
It seems that --set-string does not work with boolean as expected under helm 3.0.0-rc.3

According to https://helm.sh/docs/using_helm/#the-format-and-limitations-of-set, 
> --set-string foo=true results in a string value of "true".

Running the following, I would expect it to quote the "true", and deploy the nginx-ingress on the nodes matching the selector.
```
helm upgrade --install nginx-ingress stable/nginx-ingress --namespace nginx-ingress --set-string controller.nodeSelector."boolean\.dotted/slashed-dashed"=true
```
However, I get the following error:
```
Release "nginx-ingress" does not exist. Installing it now.
Error: Deployment in version "v1" cannot be handled as a Deployment: v1.Deployment.Spec: v1.DeploymentSpec.Template: v1.PodTemplateSpec.Spec: v1.PodSpec.NodeSelector: ReadString: expects " or n, but found t, error found in #10 byte of ...|-dashed":true},"serv|..., bigger context ...|,"nodeSelector":{"boolean.dotted/slashed-dashed":true},"serviceAccountName":"nginx-ingress-nginx-|...
```

It seems to be a regression of #2848, the code merged in #4142 has disappeared 

Output of `helm version`:
```helm version
version.BuildInfo{Version:"v3.0.0-rc.3", GitCommit:"2ed206799b451830c68bff30af2a52879b8b937a", GitTreeState:"clean", GoVersion:"go1.13.4"}
```
Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:18:23Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```
Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
Bare metal nodes deployed with RKE

### PR DESCRIPTION
**Pull request by bacongobbler**
see #3912, #4142. #4682, and #5151
closes #6940

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

### CODE REVIEW

**Comment by bacongobbler** [comment]
Looks like golangci-lint is failing to build:

> go/pkg/mod/github.com/golangci/golangci-lint@v1.21.0/pkg/golinters/wsl.go:43:6: unknown field 'AllowCaseTrailingWhitespace' in struct literal of type wsl.Configuration

Will look into this further.

**Comment by bacongobbler** [comment]
https://github.com/helm/helm/pull/6942 should fix the CI issue, but this should be reviewable on its own.

File: pkg/strvals/parser.go [hunk 1]
```diff
@@ -375,8 +380,12 @@ func typedVal(v []rune, st bool) interface{} {
 		return false
 	}
 
-	// If this value does not start with zero, and not returnString, try parsing it to an int
-	if !st && len(val) != 0 && val[0] != '0' {
+	if strings.EqualFold(val, "null") {
+		return nil
+	}
+
```
Comment (mattfarina) [inline pkg/strvals/parser.go]: In the current v2 there is also...

https://github.com/helm/helm/blob/0ca95341c19fcaf9cecdf50caef1693568835a84/pkg/strvals/parser.go#L396-L398

It's between these spaces. Should that be here, too.

Comment (bacongobbler) [reply pkg/strvals/parser.go]: yes, good catch. That was from #5151. I picked up the null fix from #3912.

**Comment by bacongobbler** [comment]
Okay, I ran through parser.go and parser_test.go to see if there were any other missing changes. I managed to find a few fixes that were not ported over from Helm 2. That should be fixed now.

File: pkg/strvals/parser.go (hunk 1 above)
Comment (bacongobbler) [reply pkg/strvals/parser.go]: fix applied.

**Review by mattfarina — APPROVED** [review]

**Pull request merged**